mangaki/mangaki/views.py

```diff
@@ -614,20 +614,23 @@ def get_works(request, category):
     return HttpResponse(json.dumps(data), content_type='application/json')
 
 
-def get_reco_list(request, category, editor):
+@login_required
+def get_reco_algo_list(request, algo, category):
     reco_list = []
-    for work, is_manga, in_willsee in get_recommendations(request.user, category, editor):
-        reco_list.append({'id': work.id, 'title': work.title, 'poster': work.ext_poster, 'synopsis': work.synopsis,
-            'category': 'manga' if is_manga else 'anime', 'rating': 'willsee' if in_willsee else None})
-    return HttpResponse(json.dumps(reco_list), content_type='application/json')
-
+    data = get_reco_algo(request.user, algo, category)
+    works = data['works']
+    for work_id in data['work_ids']:
+        work = works[work_id]
+        reco_list.append({'id': work.id, 'title': work.title, 'poster': work.ext_poster, 'synopsis': work.synopsis, 'category': work.category.slug})
+    return HttpResponse(json.dumps(reco_list, indent=4), content_type='application/json')
```

Comment:
It's actually a pain in the ass because `in_bulk` objects are not ordered. So I prefer to keep dictionaries. Maybe we will make some patch in the future.